src/components/footer.tsx

```diff
@@ -20,6 +22,13 @@ export const FooterLanguageToggle = () => {
           "button eyebrow is-small is-justify-content-center",
           locale === "en" && "is-selected"
         )}
+        onClick={() =>
+          logAmplitudeEvent("languageToggle", {
```

Comment:
small: maybe it would be useful to have another function like `logAmplitudePageEvent` or `logAmplitudeEventWithPage` that automatically grabs and appends the page info?

Reply:
Yeah, that was my initial plan but when I tried that I kept getting this error complaining about where the function was being called. I couldn't just have functions like `useLocation` (or a function like `logAmplitudeEventWithPage ` that itself called one of those functions) called directly in the JSX, but when I separated the part of getting the page data and moved it up to the first part of the functional component that worked. 

Here's what the error looked like. 
<img width="1026" alt="image" src="https://user-images.githubusercontent.com/16906516/187766087-2e11a488-cf9f-4e69-b5e4-4fef9028da73.png">

Reply:
hmm so i think the reacty way to fix this might be to make logAmplitudeEventWithPage itself a hook that returns a function you can call to log an event.... or you could skip this entirely by using window.location and messing around with that string. def just throwing around ideas though, this doesn't have to be in this PR or anything

Reply:
oh wow interesting, thanks so much, that worked!